src/classes/ACCT_AdministrativeNameRefresh_BATCH.cls

```diff
@@ -47,13 +47,18 @@ public class ACCT_AdministrativeNameRefresh_BATCH implements Database.Batchable<
     public void execute(Database.BatchableContext bc, List<Account> returnAccounts){
       	List<Account> accountsToRename = new List<Account>();
         List<Contact> contactsToProcess = new List<Contact>();
-
+        Set<Id> primaryCons = new Set<Id>(); 
+        
         if (returnAccounts.size() > 0) {
-            for (Account a : [SELECT Id, RecordTypeId,
-                             (SELECT Id, AccountId, Account.RecordTypeId, FirstName, LastName, Salutation FROM Contacts)
-                              FROM Account WHERE Id IN :returnAccounts])
-        	{
-                contactsToProcess.add(a.Contacts);
+            for (Account a : returnAccounts) {
+                if (a.Primary_Contact__c != NULL) {
+                    primaryCons.add(a.Primary_Contact__c); 
```

Comment:
@kdy618 do we do anything if we can't find a primary contact? AKA should we always expect one?

Reply:
@lifewithryan nope we currently don't do anything because we're assuming it's always going to find one.